core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -3488,6 +3488,10 @@ protected ClientTransportFactory buildTransportFactory() {
       assertThat(Throwables.getStackTraceAsString(e.getCause())).contains("deadline");
     }
 
+    // run the delayed cancel request task for deadline exceeded.
+    timer.forwardTime(5, TimeUnit.SECONDS);
```

Comment:
This test method is not concerned about the delayed cancellation of deadline. These two added lines are not needed.

Reply:
In the `@After`, it actually verifies if `timer.getDueTasks()` is empty. Since we are adding a new delayed task(send cancel to server), we need to let the task run.